### PR TITLE
applications: asset_tracker_v2: Add documentation for LwM2M integration

### PR DESCRIPTION
**Pull request by simensrostad**
Add documentation for LwM2M integration

**Rebased on https://github.com/nrfconnect/sdk-nrf/pull/7458**

### CODE REVIEW

**Comment by simensrostad** [comment]
@umapraseeda Squashed you doc commit.

File: applications/asset_tracker_v2/doc/cloud_wrapper.rst [hunk 1]
```diff
@@ -24,27 +24,87 @@ This takes place in the :file:`asset_tracker_v2/src/cloud/CMakeLists.txt` file.
...
+The following sections explain typical concepts in LwM2M and its implementation in the application.
+
+Bootstrapping and credential handling
+-------------------------------------
+
+When the option :kconfig:option:`CONFIG_LWM2M_INTEGRATION_PSK` is enabled, the modem is provisioned at run time after boot with a `Pre-Shared Key (PSK)`_ set by :kconfig:option:`CONFIG_LWM2M_INTEGRATION_PSK`.
+
+If :kconfig:option:`CONFIG_LWM2M_RD_CLIENT_SUPPORT_BOOTSTRAP` is enabled, the PSK is provisioned to a security tag dedicated to the bootstrap server connection.
+During bootstrapping, the application receives a separate newly generated key from the bootstrap server that is provisioned to a security tag dedicated to the management server connection.
+This PSK is used in the management server connection that the application connects to, after bootstrapping has been completed.
```
Comment (umapraseeda) [inline applications/asset_tracker_v2/doc/cloud_wrapper.rst:53]:
```suggestion
This PSK is used in the management server that the application connects to, after bootstrapping is completed.
```

File: applications/asset_tracker_v2/doc/cloud_wrapper.rst [hunk 2]
```diff
@@ -24,27 +24,87 @@ This takes place in the :file:`asset_tracker_v2/src/cloud/CMakeLists.txt` file.
...
+.. note::
+
+   The :kconfig:option:`CONFIG_LTE_PSM_REQ_RAT` option only increases the value that is requested by the modem from the network, not what is actually given by the network.
+
+Send operation
+--------------
+
+The application exclusively uses the `LwM2M v1.1`_ send operation to send data to the server.
+This operation enables the application to explicitly send data to the server, whenever it wants.
+The user can setup the server to enable periodic notifications for objects and resources, but there is no guarantee that the resources will change value between notifications using that approach.
```
Comment (umapraseeda) [inline applications/asset_tracker_v2/doc/cloud_wrapper.rst:90]:
```suggestion
The user can set up the server to enable periodic notifications for objects and resources, but there is no guarantee that the resources will change value between notifications using that approach.
```

**Review by umapraseeda — COMMENTED** [review]
Few more minor fixes..

**Review by SeppoTakalo — APPROVED** [review]

File: applications/asset_tracker_v2/doc/cloud_module.rst [hunk 3]
```diff
@@ -24,56 +24,68 @@ See :ref:`Cloud wrapper API <api_cloud_wrapper>` for more information on how eac
...
+| `LwM2M`_ v.1.1 compliant service (`Coiote Device Management`_, `Leshan homepage`_) |    `LwM2M`_                   |
+|                                                                                    +-------------------------------+
+|                                                                                    |    `UDP protocol`_            |
+|                                                                                    +-------------------------------+
+|                                                                                    |    `DTLS`_                    |
+|                                                                                    +-------------------------------+
+|                                                                                    |    :ref:`FOTA <nrf9160_fota>` |
+|                                                                                    +-------------------------------+
+|                                                                                    |    :ref:`lib_nrf_cloud_agps`  |
++------------------------------------------------------------------------------------+-------------------------------+
```
[the remark this answers is not in the document]
Comment (simensrostad) [reply applications/asset_tracker_v2/doc/cloud_module.rst:77]: Good suggestion, updated.

**Review by umapraseeda — APPROVED** [review]

**Review by VeijoPesonen — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**